### PR TITLE
[FVM] Fix transactions with low gas

### PR DESCRIPTION
**Pull request by janezpodhostnik**
This is a port of https://github.com/onflow/flow-go/pull/3013 to master, but I also applied some review comments from #3013

### CODE REVIEW

**Comment by pattyshack** [comment]
can you rebase / resolve the merge conflict first?

File: fvm/transactionInvoker.go [hunk 1]
```diff
@@ -151,13 +190,11 @@ func (i *TransactionInvoker) Process(
 		defer sth.EnableAllLimitEnforcements()
 
 		// drop delta since transaction failed
-		childState.View().DropDelta()
```
Comment (pattyshack) [inline fvm/transactionInvoker.go:154]: this is outdated

File: fvm/errors/errors.go [hunk 2]
```diff
@@ -69,7 +69,7 @@ func HandleRuntimeError(err error) error {
 	// if is not a runtime error return as vm error
 	// this should never happen unless a bug in the code
 	if runErr, ok = err.(runtime.Error); !ok {
-		return NewUnknownFailure(runErr)
```
Comment (pattyshack) [inline fvm/errors/errors.go:72]: this is outdated

**Comment by janezpodhostnik** [comment]
Actually let me split this up and do it a bit differently

**Pull request closed without merging**